src/libraries/System.Net.NetworkInformation/src/System/Net/NetworkInformation/NetworkAddressChange.Unix.cs

```diff
@@ -16,6 +16,12 @@ public partial class NetworkChange
         // Lock controlling access to delegate subscriptions, socket initialization, availability-changed state and timer.
         private static readonly object s_gate = new object();
 
+        // s_socketCreateContext variable was introduced to prevent event thread leaks on Linux
+        // as described on GitHub: https://github.com/dotnet/runtime/issues/63788
+        // For each CreateSocket() call we increment this variable and pass it to ".NET Network Address Change" thread.
+        // This is then used in method LoopReadSocket as additional condition to exit thread.
+        private static volatile int s_socketCreateContext;
```

Comment:
suggestion: instead of adding `s_socketCreateContext` you can put the socket handle in a reference type.

```cs
class ListenerSocket { public ListenerSocket(int handle); public int Handle; }
static volatile ListenerSocket? s_socket;
```

Reply:
Hi,

Thank you for this suggestion! Seems more elegant solution, I changed this with following commit: https://github.com/dotnet/runtime/pull/63963/commits/e130fdcfa4eb9ce6552d7c9278cd92aaf2a5f383

Reply:
Looks good.
You can remove `SocketWrapper.NotSet` and use `null` instead.

Reply:
Initially I had nullable implementation, but then changed it to non-nullable one because I don't like nulls :)

Don't know what is better here ... 

:)

Reply:
`null` is better. It is a compile-time constant. It's also the default value for fields, and it's a good value to indicate _not set_.